File: ALGO/ECI 2017 1 ALGO Final Test/6823.py
from sys import stdin,stdout
#Sergio Andres Rodriguez Torres
#Veredicto: Time limit exceeded
def main():
    v = [0]*3 
    for l in stdin.readlines():
        l = l.strip()
        c = v[1] = v[2] = 0
        v[0] = 1
        r = 0
        for i in l:
            if not i.isdigit():
                c = v[1] = v[2] = 0
                v[0] = 1
                continue
            c = (c + int(i))%3
            r += v[c]
            v[c] += 1
        print(r)
main()

File: ALGO/ECI 2017 1 ALGO Final Test/C++/6823.cpp
#include <bits/stdc++.h>
using namespace std;
//Sergio Andres Rodriguez Torres
//Veredicto: Accepted
int v[3];
int main(){
    ios_base::sync_with_stdio(false);
    string a;
    long long ans;
    while(cin>>a){
        int c = 0;
        v[1] = v[2] = 0;
        v[0] = 1;
        ans = 0;
        for(int i = 0; i < a.length(); i++){
            if(!isdigit(a[i])){
                c = 0;
                v[1] = v[2] = 0;
                v[0] = 1;
                continue;
            }
            c = (c+a[i]-'0')%3;
            ans += v[c];
            v[c]++;
        }
        cout<<ans<<endl;
    }
}
